cassandra-scheduler/src/test/java/io/mesosphere/mesos/frameworks/cassandra/PersistedCassandraClusterHealthCheckHistoryTest.java

```diff
@@ -0,0 +1,412 @@
+
+        hc2.setHealthy(true);
+        hc2.setMsg("msg");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(hc1.build(), hc2.build()));
+
+        hc2.clearMsg();
+        hc1.setMsg("msg");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(hc1.build(), hc2.build()));
+
+        hc1.clearMsg();
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(hc1.build(), hc2.build()));
+
+    }
+
+    @Test
+    public void testIsSimilarEntryNodeInfo() throws Exception {
+        CassandraFrameworkProtos.NodeInfo.Builder ni1 = CassandraFrameworkProtos.NodeInfo.newBuilder();
+        CassandraFrameworkProtos.NodeInfo.Builder ni2 = CassandraFrameworkProtos.NodeInfo.newBuilder();
+
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+        ni1.setUptimeMillis(1234);
+        ni2.setUptimeMillis(5678);
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.setClusterName("cluster");
+        ni2.setClusterName("cluster");
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.setNativeTransportRunning(true);
+        ni2.setNativeTransportRunning(true);
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni2.setNativeTransportRunning(false);
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni2.setNativeTransportRunning(true);
+        ni2.setClusterName("foo");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni2.setNativeTransportRunning(true);
+        ni2.setClusterName("cluster");
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.setDataCenter("dc");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+
+        ni1.clearDataCenter();
+        ni2.setRack("rac");
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.isSimilarEntry(ni1.build(), ni2.build()));
+    }
+
+    @Test
+    public void testObjEquals() throws Exception {
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.objEquals(null, null));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals(null, ""));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals("", null));
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.objEquals("1", "1"));
+        assertTrue(PersistedCassandraClusterHealthCheckHistory.objEquals(1, 1));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals("1", "2"));
+        assertFalse(PersistedCassandraClusterHealthCheckHistory.objEquals(1, 2));
+    }
```

Comment:
I have another test (below) to add that fails.

Mesos framework messages aren't guaranteed to be delivered in order, so the logic in `record` will need to be updated to account for this fact.

```java
    @Test
    public void testAllOscillatingOutOfOrder() throws Exception {
        State state = new InMemoryState();
        PersistedCassandraClusterHealthCheckHistory hcHistory = new PersistedCassandraClusterHealthCheckHistory(state);

        hcHistory.record("abc", 10, unhealthy());
        hcHistory.record("abc", 11,   healthy());
        hcHistory.record("abc", 13, unhealthy());
        hcHistory.record("abc", 16, unhealthy());
        hcHistory.record("abc", 14,   healthy());
        hcHistory.record("abc", 19, unhealthy());
        hcHistory.record("abc", 15,   healthy());
        hcHistory.record("abc", 12,   healthy());
        hcHistory.record("abc", 25, unhealthy());
        hcHistory.record("abc", 17,   healthy());
        hcHistory.record("abc", 20,   healthy());
        hcHistory.record("abc", 21,   healthy());
        hcHistory.record("abc", 22, unhealthy());
        hcHistory.record("abc", 24,   healthy());
        hcHistory.record("abc", 18,   healthy());
        hcHistory.record("abc", 23,   healthy());

        final CassandraClusterHealthCheckHistory history = hcHistory.get();
        final List<CassandraFrameworkProtos.HealthCheckHistoryEntry> list = history.getEntriesList();

        CassandraFrameworkProtos.HealthCheckHistoryEntry prev = null;
        for (final CassandraFrameworkProtos.HealthCheckHistoryEntry entry : list) {
            if (prev != null) {
                assertThat(entry.getTimestampFirst()).isLessThanOrEqualTo(entry.getTimestampLast());
                assertThat(prev.getTimestampLast()).isLessThanOrEqualTo(entry.getTimestampFirst());
            }
            prev = entry;
        }
    }

    static HealthCheckDetails healthy() {
        return healthCheckDetails(true);
    }
    static HealthCheckDetails unhealthy() {
        return healthCheckDetails(false);
    }

    static HealthCheckDetails healthCheckDetails(final boolean healthy) {
        return HealthCheckDetails.newBuilder()
            .setHealthy(healthy)
            .setInfo(CassandraFrameworkProtos.NodeInfo.newBuilder()
                .setClusterName("cluster")
                .setDataCenter("dc1")
                .setUptimeMillis(1).build()).build();
    }
```